diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt

```diff
@@ -140,36 +140,45 @@ class NullChecksRule(configRules: List<RulesConfig>) : DiktatRule(
                         |${elseCodeLines.joinToString(separator = "\n")}
                         |} ?: break
                         """.trimMargin()
-                else -> """
-                        |$variableName?.let {
-                        |${elseCodeLines.joinToString(separator = "\n")}
-                        |}
-                        |?: run {
-                        |${thenCodeLines.joinToString(separator = "\n")}
-                        |}
-                    """.trimMargin()
+                else -> getDefaultCaseNullCheck(variableName, elseCodeLines, thenCodeLines)
             }
         } else {
             when {
                 elseCodeLines.isNullOrEmpty() || (elseCodeLines.singleOrNull() == "null") ->
                     "$variableName?.let {${thenCodeLines?.joinToString(prefix = "\n", postfix = "\n", separator = "\n")}}"
                 elseCodeLines.singleOrNull() == "break" ->
                     "$variableName?.let {${thenCodeLines?.joinToString(prefix = "\n", postfix = "\n", separator = "\n")}} ?: break"
-                else -> """
-                        |$variableName?.let {
-                        |${thenCodeLines?.joinToString(separator = "\n")}
-                        |}
-                        |?: run {
-                        |${elseCodeLines.joinToString(separator = "\n")}
-                        |}
-                    """.trimMargin()
+                else -> getDefaultCaseNullCheck(variableName, thenCodeLines, elseCodeLines)
             }
         }
         val tree = KotlinParser().createNode(text)
         condition.treeParent.treeParent.addChild(tree, condition.treeParent)
         condition.treeParent.treeParent.removeChild(condition.treeParent)
     }
 
+    private fun getDefaultCaseNullCheck(
+        variableName: String,
+        thenCodeLines: List<String>?,
+        elseCodeLines: List<String>?
+    ): String {
+        val editedThenPart = if (variableName == thenCodeLines?.singleOrNull()) {
```

Comment:
Have you considered counting statements in `BLOCK_EXPRESSION` nodes? It will be more accurate, than counting lines

Reply:
```
if (result != null) {
    goo()
} else {
    for(i in 1..10)
        break
}
```
In this example node else has only one child statement and this is why we should count all the statements in block (maybe recursively)
Is it really what we want?

Reply:
Is this the check whether to wrap it into `run`? If yes, then even multi-line expression shouldn't be wrapped, so we don't need to check recursively.

Reply:
It is a check if we need to put it in `run` block or just after `?:`

Reply:
Yes, so your example should be
```kotlin
result?.let {
    goo()
}
     ?: for(i in 1..10)
        break
```

Reply:
Ok, thanks